substrate/bin/node/testing/src/keyring.rs

```diff
@@ -75,6 +75,7 @@ pub fn session_keys_from_seed(seed: &str) -> SessionKeys {
 /// Returns transaction extra.
 pub fn tx_ext(nonce: Nonce, extra_fee: Balance) -> TxExtension {
 	(
+		frame_system::AuthorizeCall::new(),
```

Comment:
I find it a little bit weird that such a fundamental feature requires that the user enables this special transaction extension. 
But yeah discussable.

Reply:
This format with the `AuthorizeCall` extension and the `#[pallet::authorize]` macro is just one way of doing transactions without signatures (I'm trying to not call them "unsigned" anymore :grin: ). You don't necessarily need to use `AuthorizeCall`, you could use your own extension, but we want to provide this mechanism because it's cleaner and easier for users to just put some logic in that closure, just like they were previously implementing `ValidateUnsigned`.

Also, realistically if we put this in our runtime extension pipelines and examples, most people will do the same and just use the feature. Once this is merged, we will also communicate how to switch from old unsigned transactions to this new format.

Reply:
But we could warn or fail at runtime: the transaction extension could set a storage value `AuthorizeCallExecuted` to `true`, and `note_applied_extrinsic` could remove it.

It is a bit of a hack, but we are using frame-system storage to store information about the current extrinsic, so this behavior would not break anything.